### ISSUE TITLE
File "adafruit_ble_heart_rate.py", line 179, in measurement_values RuntimeError: buffer too small

### ISSUE THREAD

**Issue by dglaude** [adafruit/Adafruit_CircuitPython_BLE_Heart_Rate#3]
Running (a modified version of) https://learn.adafruit.com/circuitpython-ble-heart-rate-monitor-gizmo I do get the following error after connection and getting a few reading OK:

```
Traceback (most recent call last):
  File "main.py", line 120, in <module>
  File "adafruit_ble_heart_rate.py", line 179, in measurement_values
RuntimeError: buffer too small
```

This is running 5.0.0-beta.5 and https://github.com/adafruit/Adafruit_CircuitPython_BLE_Heart_Rate/releases/tag/1.0.0 + https://github.com/adafruit/Adafruit_CircuitPython_BLE/releases/tag/5.0.3 from adafruit-circuitpython-bundle-5.x-mpy-20200206

The error occurs on the line with hr_service.measurement_values:
```
        while hr_connection.connected:
            values = hr_service.measurement_values
            print(values)  # returns the full heart_rate data set

```

Here are the kind of values I would get when it does not crash:

> HeartRateMeasurementValues(heart_rate=87, contact=True, energy_expended=None, rr_intervals=[652, 667, 665])
> HeartRateMeasurementValues(heart_rate=88, contact=True, energy_expended=None, rr_intervals=[801, 791])
> HeartRateMeasurementValues(heart_rate=86, contact=True, energy_expended=None, rr_intervals=[764])

I will try with bigger buffer, using the py file rather than the mpy:
```
#### TODO Use commented line instead once .packet_size is fixed in CircuitPython beta.
####    self._measurement_buf = bytearray(self.heart_rate_measurement.packet_size)
#### TEMPORARY
        self._measurement_buf = bytearray(20)
```

Maybe I do get more that 3 rr data? (whatever that mean) and 20 is not enough for my device?

I will try to report on success or failure.

**Comment by dglaude** [comment]
I can reproduce with the .py file replacing the .mpy file and using ble_heart_rate_simpletest.py example:

```
Traceback (most recent call last):
  File "main.py", line 55, in <module>
  File "/lib/adafruit_ble_heart_rate.py", line 179, in measurement_values
RuntimeError: buffer too small
```

**Comment by dglaude** [comment]
Not happening anymore... the only change is that I unplugged the board, and retried in another location a bit later.

I do get glitch result, so maybe my cheap heat rate monitor is sending bad data from time to time (but I should be able to filter it out those data):

https://gist.github.com/dglaude/dde2967b6a422e5a65db1089218d871c

Maybe this should be closed, I will try to debug the raw data that I get from that hardware (and I have a second one exactly the same).

**Comment by dhalbert** [comment]
```
#### TODO Use commented line instead once .packet_size is fixed in CircuitPython beta.
####    self._measurement_buf = bytearray(self.heart_rate_measurement.packet_size)
#### TEMPORARY
        self._measurement_buf = bytearray(20)
```
This is fixed in 5.0.0 beta.5, you should be be able to uncomment the `.packet_size` line and remove the `(20)` line.

**Comment by kattni** [comment]
Appears to b resolved. Closing this issue.
